pkg/test/echo/server/forwarder/instance.go

```diff
@@ -152,13 +156,16 @@ func (i *Instance) Run(ctx context.Context) (*proto.ForwardEchoResponse, error)
 	case <-ctx.Done():
 		responsesMu.RLock()
 		defer responsesMu.RUnlock()
-		c := 0
-		for _, res := range responses {
-			if res != "" {
+		var c int
+		var tt time.Duration
+		for id, res := range responses {
+			if res != "" && responseTimes[id] != 0 {
 				c++
+				tt += responseTimes[id]
 			}
 		}
-		return nil, fmt.Errorf("request set timed out after %v and only %d/%d requests completed", i.timeout, c, i.count)
+		avgTime := tt / time.Duration(c)
```

Comment:
should you protect against dividing by 0?

Reply:
good catch